model/HintList.ts

```diff
@@ -4,31 +4,17 @@
  * of the Wikipedia page of the celebrity.
  */
 export class HintList {
-    birth: Hint<Date> ;
-    ageOrDeath : Hint<Date | number> ;
-    occupation : Hint<string> ;
-    citizenship : Hint<string> ;
-    initials : Hint<string> ;
-    paragraph : Hint<string> ;
+
+    list: Hint<any>[] = [];
 
     constructor(
-        birth: Date | undefined ,
-        ageOrDeath: Date | number | undefined,
-        occupation: string | undefined,
-        citizenship: string | undefined,
-        initials: string | undefined,
-        paragraph: string | undefined
+        ...params: Hint<any>[]
     ){
-        this.birth = new Hint(1,  true, birth)
-        this.ageOrDeath = new Hint(1, true, ageOrDeath);
-        this.occupation = new Hint(1, true, occupation);
-        this.citizenship = new Hint(1, true, citizenship);
-        this.initials = new Hint(2, true, initials);
-        this.paragraph = new Hint(2, true, paragraph);
+        this.list = params
     }
 
     toList() : Hint<any>[]{
-        return [this.birth, this.ageOrDeath, this.occupation, this.citizenship, this.initials, this.paragraph] ;
+        return this.list ;
```

Comment:
This won't work ! We should be able to link a hint with what the hint is. Herre how can we know if a certain date hint is the birth or death ?

Reply:
If you want, we can put some hints as "compulsory", for instance the `birthDate`, the `deathDate`, the initials and the first paragraph. But we should also allow an arbitrary number of other hints we don't know by advance as parameters to accept hints specific to a type of personality, as we discussed yesterday. For instance, our `HintList` should accept a hint for Nobel prizes for scientists, a hint for NBA teams for basketball players, or a hint for discography for singers. Because this list must be generic and the same for every personalities, we need "non-compulsory" arbitrary hints. What do you think about that ?

Reply:
I agree but we still need to label these non-compulsory hints. My current idea is to add a label : string element to the hint class

Reply:
having this arbitrary list without changing the Hint class is not a good idea in my mind

Reply:
We can modify the `Hint` class as well. I agree with having a kind of label for each hint. But do you agree with having some compulsory hints and some arbitrary hints ?